test/join.jl

```diff
@@ -578,4 +578,54 @@ module TestJoin
         @test eltypes(join(l, r, on=:b, kind=:outer, makeunique=true)) ==
             [Union{Int, Missing}, CS, Union{Int, Missing}]
     end
+
+    @testset "test checks of merge key uniqueness" begin
+        @test_throws ArgumentError join(name, job, on=:ID,
+                                        validate=(false, true))
+        @test_throws ArgumentError join(name, job, on=:ID,
+                                        validate=(true, true))
+        @test_throws ArgumentError join(job, name, on=:ID,
+                                        validate=(true, false))
+        @test_throws ArgumentError join(job, name, on=:ID,
+                                        validate=(true, true))
+        @test_throws ArgumentError join(job, job, on=:ID,
+                                        validate=(true, true))
+
+
+        @test join(name, job, on=:ID, validate=(true, false)) ==  inner
+        @test join(name, job, on=:ID, kind=:inner, validate=(false, false)) == inner
+
+        # Make sure ok with various weird characters
```

Comment:
These are not "characters". And "weird" could just be "special".

Below, it would be useful to include a `0` in the values of `ID`, to check that it's treated as different from `-0.0`. Then you can drop `0.0` from the list here. Without this, the code doesn't actually test that `0.0` and `-0.0` are considered as different.